src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,58 +1009,160 @@ public static void onWorldLoad(@NotNull final World world)
+                addToZipFile(getSaveLocation(), zos);
 
-            zos.close();
-            fos.close();
+                zos.close();
+            }
+            catch (final Exception e)
+            {
+                /**
+                 * Intentionally not being propagated.
+                 */
+                Log.getLogger().warn("Unable to backup colony data, please contact an administrator");
+                Log.getLogger().debug("Exception:", e);
+                // Don't leave a mess - release the token before we return
+                backupInProgress = false;
+                return false;
+            }
+            // remove any surplus backups
+            pruneOldBackups(Configurations.administration.colonyBackupsToKeep);
+            // Schedule the next backup
+            scheduleColonyBackup();
+            // Release the token before we return
+            // no need to be thread safe here, we're the only ones who have it
+            backupInProgress = false;
+
+            return true;
         }
-        catch (final Exception e)
+        else
         {
-            /**
-             * Intentionally not being thrown.
-             */
-            Log.getLogger().warn("Unable to backup colony data, please contact an administrator");
+            // We were denied the backup token. That means there's
+            // another thread already in the process of running the backup
             return false;
         }
-
-        return true;
     }
 
-    public static void addToZipFile(final String fileName, final ZipOutputStream zos, final File folder)
+    public static void addToZipFile(final File file, final ZipOutputStream zos)
     {
-        final File file = new File(folder, fileName);
-        try(FileInputStream fis = new FileInputStream(file))
+        try (FileInputStream fis = new FileInputStream(file))
         {
-            zos.putNextEntry(new ZipEntry(fileName));
+            zos.putNextEntry(new ZipEntry(file.getName()));
             Files.copy(file, zos);
-            fis.close();
         }
         catch (final Exception e)
         {
             /**
              * Intentionally not being thrown.
              */
-            Log.getLogger().warn("Error packing " + fileName + " into the zip.");
+            Log.getLogger().warn("Error packing " + file.getName() + " into the zip.");
+            Log.getLogger().debug(e);
+        }
+    }
+
+    /**
+     * Removes oldest backup files to reduce kept number as desired
+     */
+    private static void pruneOldBackups(final int numToKeep)
+    {
+        if (numToKeep > 0)
+        {
+            final File[] backupFiles = getBackupSaveDirectory().listFiles();
+            if (backupFiles != null && backupFiles.length > numToKeep)
+            {
+                Arrays.sort(backupFiles);
+                for (int i = 0; i < backupFiles.length - numToKeep; ++i)
+                {
+                   if(!backupFiles[i].delete())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this line start at column 21. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)